tools/analyse_query_usage.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Usage:
+#
+#   ./analyse_query_usage.sh my-file-of-urls.txt
+#
+#   cat my-file-of-urls.txt | ./analyse_query_usage.sh
+#
+#
+# For the input, a list of URLs, shows which are most common when you
+# ignore all query parameters. Eg, those that will be normalised in
+# mappings unless some query parameters are marked as important.
+#
+# Eg:
+#   162 http://www.planningportal.gov.uk/PpPortalSupport/ChangeLanguage
+#    45 http://www.planningportal.gov.uk/PpWeb/jsp/redirect.jsp
+#       ...
+#
+
+# Input is either file arg contents or stdin
+cat "$@" | \
+  # Take all of the URL leading up to `?`, or all if not present
+  cut -d? -f1 | \
```

Comment:
Doing that on the input to this script would mean stripping out trailing slashes from the middle of the URL (because the input contains query strings) rather than dropping them from the ends of lines here. You're right, though - there are other changes made by canonicalization which this script shouldn't need to make (downcasing, encoding etc) so it makes sense to not do any of them here.